### ISSUE TITLE
Feasibility of Running Code on GPU

### ISSUE THREAD

**Issue by Pourbahman** [canjiali/PARADE#18]
Hi,

Thanks for all of your complete answers in other issues. Also, thanks for your awesome work shared.
 
I used the following config:
number of epoch = 1
number of fold = 1
model checkpoint = bert_mini (uncased_L_4_H_256_A_4)
batch_size = 64

The memory consumption is as follows:
CPU RAM = 80G
GPU RAM = 6.5G

**Your model is training after about 10 hours for just 1 epoch and 1 fold and it did not finish**

**I changed batch_size to 128** and then the memory consumption is as follows:
CPU RAM = 120G
GPU RAM = 6.5G

**Your model is training after about 14 hours for just 1 epoch and 1 fold and up to now did not finish**

Would you please tell me are you sure I can train your code on GPU? (according to #14)

Thanks in advance

**Comment by canjiali** [comment]
Do you mean that you're using CPU for model training? also, I don't think you can use a batch size of 128.

You can also check the capreolus implementation for GPU. https://github.com/capreolus-ir/capreolus/blob/master/docs/reproduction/PARADE.md

**Comment by Pourbahman** [comment]
I checked it as you told but unfortunately, Andrew told a TPU or GPU(s) with 48GB RAM are really needed to run this with reasonable hyperparameters. 

Colab Pro version has tpu v.2 and keep run for 12 hours. Is it possible to run your code by it? (I mean after each 12 hours I save loss function, model, ... and then again load it from that check point and continue running? Do you think it is possible?

If yes, would you please guide me these changes where in your code should be applied?

Thanks in advance

**Comment by canjiali** [comment]
you should be able to run one fold within 12 hours on a tpu v2. so you don't need to worry about reloading the checkpoint.

**Comment by Pourbahman** [comment]
Thanks. 

1. If I do not have had gcp, can I use google drive to load data and save the result?

2. Also, which changes should I do in your code where I can use the Colab pro using my Google drive?

3. Also, folds are evaluated completely separate? and are independent?

**Comment by canjiali** [comment]
1. maybe, you can google for it
2. you can generate the training/testing data on your local machine, upload the google drive, then train the model. you can paste the code there and run the code......
3. predicted seperately. merged for evaluation.

**Comment by Pourbahman** [comment]
Thank you.

In which model of BERT I can run each fold less than 12 hours?
